numpy/core/_internal.py

```diff
@@ -756,3 +756,12 @@ def _ufunc_doc_signature_formatter(ufunc):
         out_args=out_args,
         kwargs=kwargs
     )
+
+
+def _is_from_ctypes(obj):
+    # determine if an object comes from ctypes, in order to work around
+    # a bug in the buffer protocol for those objects, bpo-10746
+    try:
+        return 'ctypes' in type(obj).__mro__[-2].__module__
```

Comment:
`__mro__` is only available for new style classes, might be a problem in Python 2.7.

Reply:
Although I suspect anything that inherits from ctypes is going to be a new style class, so this should be OK. Maybe a note?

Reply:
Yep, it seemed that ctypes classes are always new-style